test/framework/src/main/java/org/elasticsearch/test/rest/TestFeatureService.java

```diff
@@ -8,10 +8,8 @@
 
 package org.elasticsearch.test.rest;
 
-import java.util.Set;
-
 public interface TestFeatureService {
-    boolean clusterHasFeature(String featureId);
+    TestFeatureService ALL_FEATURES = feature -> true;
 
-    Set<String> getAllSupportedFeatures();
+    boolean clusterHasFeature(String featureId);
```

Comment:
This could just be a `Predicate<String>`, but we maybe just want to keep it as an explicit interface...

Reply:
IMHO, the value of extending Predicate is rather limited here.
We sure want to to keep an explicit interface. However, `Predicate#and` returns a `Predicate`, so it's not of much use in this case. Sometimes it'd be nice to have type aliases.